taskvine/src/worker/vine_worker.c

```diff
@@ -320,6 +326,11 @@ static void measure_worker_resources()
 
 	if (manual_disk_option > 0) {
 		r->disk.total = MIN(r->disk.total, manual_disk_option);
+	} else {
+		/* Set the reporting disk to a fraction of the measured disk to avoid
+		 * unnecessarily forsaking tasks with unspecified resources. */
+		manual_disk_option = floor(r->disk.total * disk_percent / 100);
```

Comment:
Protect for values outise [0,100]:

r->disk.total = ceil(r->disk.total * MIN(100, MAX(disk_percent, 0)));

and let's leave manual_disk_option unset to mark that the disk was not specified from the command line.